### ISSUE TITLE
rule propertydefinition separator conflicting with vs201x Separator

### ISSUE THREAD

**Issue by WorldofBay** [premake/premake-core#1157]
The optional propertydefinition field "separator" for list properties is conflicting with the optional vs201x list Separator.

They do different things but you can only use both or none.

Propertydefinition separator:
That field is the string used to write the table down so that it is read as table. For vs it needs to be a ';', but the default is ' '. So to use the list functionality in vs it has to be set. **This is not a conversion** (could be abused to be though).

Vs list Separator:
This field is used to concatenate lists to pass them as input. **This is a conversion** to command-line arguments.

Clarification:
We have a cumstom build tool, let's call it foo. Foo can have includes, using the -I switch. Foo can have multiple -I options specified. The type of -I parameters is "string".

We want to call foo -Ibar -Ibat.
We use vs201x (language c++), have a proper premake environment and a custom rule "foo" containing the following:
```lua
propertydefinition {
  name = "Includes",
  kind = "list",
  switch = "-I",
}
```
We have a project that calls foo like that:
```lua
rules "foo"
fooVars { Includes = { "bar", "bat" } }
```

This results in the following lines in the .vcxproj:
```xml
<foo>
  <Includes>bar bat</Includes>
</foo>
```

This is invalid code and will not be read by VS, because arrays are separated by ';'.
Foo on file "file.foo" is called like that:
`foo file.foo`

Now we change our property:
```lua
propertydefinition {
  name = "Includes",
  kind = "list",
  switch = "-I",
  separator = ";",
}
```

This results in the following lines in the .vcxproj:
```xml
<foo>
  <Includes>bar;bat</Includes>
</foo>
```

Valid!

But we have a new problem, now the foo.xml caintains the following code:
```xml
    <StringListProperty
      Name="Includes"
      HelpContext="0"
      Switch="-I"
      Separator=";"
      />
```
And foo is called like that:
`foo -Ibar;bat file.foo`

How do we call foo correctly?
Scrap lists and use strings in the project:
```lua
rules "foo"
fooVars { Includes = table.concat({ "bar", "bat" },";") }
```

But with that we can only add items once. Every subsequent call will use the wrong separator at some point.

So how to fix that?
Easiest way would be to change the name of the vs2010 module looks for from "separator" to "Separator" (as vs syntax is uppercase anyways) or a more verbose "vsseparator". I don't know how many projects break with that change but i'm sure there are more that finally work.
I could do it myself but i want to gather some opinion first. Maybe instead change the name of the rule separator to something looking truly internal or support modules in there?

**Comment by WorldofBay** [comment]
any thoughts on this? why is separator evaluated in the project file generation at all?

**Comment by samsinsane** [comment]
Does #1160 resolve this?

**Comment by WorldofBay** [comment]
Partly as it was a workaround.

Property lists can be used for VS now with all their features as long as you don't touch the separator which is fine for probably 99% of the usage.

**Comment by samsinsane** [comment]
Fair enough, I'm happy to leave this open if that would be desired? I think that 99% is good enough, but 100% is better haha.

**Comment by WorldofBay** [comment]
I'm currently thinking about entirely removing the functions rule.getPropertyString and rule.expandString in rule.lua.

Their only purpose is the creation of a string out of the rule property but they don't really know what the string is for.
Those two are called only out of the gmake2/vstudio module as those are the only modules with rules and never in the core (except one calling the other for one specific case, see later) and they are 2 functions for 2 modules because the modules want different things.

So i think it is better to do the string creation entirely in the respective module and not in the core. The modules know what the string is for (unlike the core) and therefore should be able to do it easier. There also is almost no abstraction or versatility provided by them, the only case where code is reused is if gmake2 has a list property without a switch and even then default behaviour is different between gmake2 and vstudio.

Long story short there is no benefit by rule.getPropertyString and rule.expandString but there is the problem i reported in this issue, so remove them.

Any objections? If not i'll move the code into the modules directly and see if i can keep any generic case in the core.

**Comment by WorldofBay** [comment]
Everything done 👍
